src/main/kotlin/net/atos/zac/zoeken/IndexingService.kt

```diff
@@ -0,0 +1,333 @@
+        if (idsToBeDeleted.isEmpty()) {
+            return
+        }
+        try {
+            solrClient.deleteById(idsToBeDeleted)
+        } catch (solrServerException: SolrServerException) {
+            throw IndexingException(solrServerException)
+        } catch (ioException: IOException) {
+            throw IndexingException(ioException)
+        }
+    }
+
+    private fun removeFromSolrIndex(id: String) {
+        try {
+            solrClient.deleteById(id)
+        } catch (solrServerException: SolrServerException) {
+            throw IndexingException(solrServerException)
+        } catch (ioException: IOException) {
+            throw IndexingException(ioException)
+        }
+    }
+
+    private fun logProgress(objectType: ZoekObjectType, voortgang: Long, grootte: Long) {
+        logTypeMessage(objectType, "reindexed: $voortgang / $grootte")
+    }
+
+    private fun removeEntitiesFromSolrIndex(objectType: ZoekObjectType) {
+        val query = SolrQuery("*:*")
+        query.setFields("id")
+        query.addFilterQuery("type:$objectType")
+        query.addSort("id", SolrQuery.ORDER.asc)
+        query.setRows(SOLR_MAX_RESULT)
+        var cursorMark = CursorMarkParams.CURSOR_MARK_START
+        var done = false
+        while (!done) {
+            query.set(CursorMarkParams.CURSOR_MARK_PARAM, cursorMark)
+            val response: QueryResponse
+            try {
+                response = solrClient.query(query)
+            } catch (solrServerException: SolrServerException) {
+                throw IndexingException(solrServerException)
+            } catch (ioException: IOException) {
+                throw IndexingException(ioException)
+            }
+            removeFromSolrIndex(
+                response.results
+                    .mapNotNull { document: SolrDocument -> document["id"] }
+                    .map { obj: kotlin.Any -> obj.toString() }
+            )
+            val nextCursorMark = response.nextCursorMark
+            if (cursorMark == nextCursorMark) {
+                done = true
+            } else {
+                cursorMark = nextCursorMark
+            }
+        }
+    }
+
+    private fun reindexAllZaken() {
+        val listParameters = ZaakListParameters()
+        listParameters.ordering = "-identificatie"
+        listParameters.page = ZGWApiService.FIRST_PAGE_NUMBER_ZGW_APIS
+        var hasMore: Boolean
+        do {
+            hasMore = reindexZaken(listParameters)
+            listParameters.page += 1
+        } while (hasMore)
+    }
+
+    private fun reindexAllInformatieobjecten() {
+        val listParameters = EnkelvoudigInformatieobjectListParameters()
+        listParameters.page = ZGWApiService.FIRST_PAGE_NUMBER_ZGW_APIS
+        var hasMore: Boolean
+        do {
+            hasMore = reindexInformatieobjecten(listParameters)
+            listParameters.page += 1
+        } while (hasMore)
+    }
+
+    private fun reindexAllTaken() {
```

Comment:
Looks a bit clumsy? Maybe Copilot has a nice suggestion to improve this code? ;-)